tensorzero-core/src/inference/types/mod.rs

```diff
@@ -993,32 +993,125 @@ pub enum ThoughtSummaryBlock {
 }
 
 /// Struct that represents a model's reasoning
-#[derive(ts_rs::TS, Clone, Debug, Deserialize, PartialEq, Serialize, JsonSchema)]
-#[ts(export)]
+#[derive(ts_rs::TS, Clone, Debug, PartialEq)]
+#[ts(export, optional_fields)]
 #[cfg_attr(feature = "pyo3", pyclass(get_all))]
 #[export_schema]
 pub struct Thought {
-    #[ts(optional)]
     pub text: Option<String>,
     /// An optional signature - currently, this is only used with Anthropic,
     /// and is ignored by other providers.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    #[ts(optional)]
     pub signature: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    #[ts(optional)]
     pub summary: Option<Vec<ThoughtSummaryBlock>>,
     /// When set, this `Thought` block will only be used for providers
     /// matching this type (e.g. `anthropic`). Other providers will emit
     /// a warning and discard the block.
-    #[serde(
-        rename = "_internal_provider_type",
-        skip_serializing_if = "Option::is_none"
-    )]
-    #[ts(optional)]
     pub provider_type: Option<String>,
 }
 
+// Custom deserializer to handle both `provider_type` and legacy `_internal_provider_type` fields,
+// allowing round-tripping of our own serialized output which includes both fields.
+// TODO(#5001): The old field is deprecated in 2025.12.1; we can remove the deserializer in 2026.2+.
```

Comment:
`_internal_provider_type` is stored in the database so this can never be removed. Might as well use `serde(alias)` here instead of a custom deserializer.